packages/core/src/ChatHistory.ts

```diff
@@ -77,20 +84,23 @@ export class SummaryChatHistory extends ChatHistory {
   summaryPrompt: SummaryPrompt;
   llm: LLM;
   private messagesBefore: number;
+  readonly #maxTokens: number;
 
-  constructor(init?: Partial<SummaryChatHistory>) {
+  constructor(
+    init?: Partial<SummaryChatHistory> & Partial<SummaryChatHistoryOptions>,
+  ) {
     super();
     this.messages = init?.messages ?? [];
     this.messagesBefore = this.messages.length;
     this.summaryPrompt = init?.summaryPrompt ?? defaultSummaryPrompt;
     this.llm = init?.llm ?? new OpenAI();
-    if (!this.llm.metadata.maxTokens) {
+    if (!this.llm.metadata.maxTokens || !init?.maxTokens) {
```

Comment:
Should we use `Infinity` if it's undefined?

Reply:
the code below won't work if maxTokens is not specified, so I think it's ok, to throw this error, it's basically a configuration error.

this class also needs a tokenizer to work correctly (and for ollama it could be any LLM with any tokenizer). 

So I think we should set `tokenizer` to `globalsHelper.defaultTokenizer.encode` only if the LLM is OpenAI and throw an error if `tokenizer` is not explicitly specified otherwise